cli/tools/compile.rs

```diff
@@ -97,33 +101,73 @@ pub async fn compile(
   );
   validate_output_path(&output_path)?;
 
-  let mut file = std::fs::File::create(&output_path)
-    .with_context(|| format!("Opening file '{}'", output_path.display()))?;
-  let write_result = binary_writer
-    .write_bin(
-      &mut file,
-      eszip,
-      &module_specifier,
-      &compile_flags,
-      cli_options,
+  let mut temp_filename = output_path.file_name().unwrap().to_owned();
+  temp_filename.push(format!(
+    ".tmp-{}",
+    faster_hex::hex_encode(
+      &rand::thread_rng().gen::<[u8; 8]>(),
+      &mut [0u8; 16]
     )
-    .await
-    .with_context(|| format!("Writing {}", output_path.display()));
-  drop(file);
-  if let Err(err) = write_result {
-    // errored, so attempt to remove the output path
-    let _ = std::fs::remove_file(output_path);
-    return Err(err);
+    .unwrap()
+  ));
+  let temp_path = output_path.with_file_name(temp_filename);
+
+  let mut file = std::fs::File::create(&temp_path).with_context(|| {
+    format!("Opening temporary file '{}'", temp_path.display())
+  })?;
+  let write_result = if compile_flags.eszip {
+    // TODO: write npm vfs
+    let (_, _, node_modules) = binary_writer.pack_node_modules(&mut eszip)?;
+    if let Some(node_modules) = node_modules {
+      eszip.add_opaque_data(
+        "internal://node_modules".to_string(),
+        Arc::from(deno_core::serde_json::to_string(&node_modules)?.as_bytes()),
+      );
+    }
+    file.write_all(&eszip.into_bytes()).map_err(AnyError::from)
```

Comment:
This is somewhat abusing the deno compile subcommand to create an eszip - functionality that isn't necessary for SSH. Could you instead move this to the test?